packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @fileoverview Hint to validate if the doctype is correct
+ */
+
+import { Category } from 'hint/dist/src/lib/enums/category';
+import { HintScope } from 'hint/dist/src/lib/enums/hintscope';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, HintMetadata, FetchEnd, ProblemLocation } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+import { MatchInformation } from './types';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class implements IHint {
+
+    public static readonly meta: HintMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `This hint checks if the HTML is using the most modern DOCTYPE.`
+        },
+        id: 'doctype',
+        schema: [],
+        scope: HintScope.any
+    }
+
+    public constructor(context: HintContext) {
+        const regExpFactory = (isFlexible = false) => {
+            return new RegExp(`(<!doctype\\s+(html)\\s*(\\s+system\\s+"about:legacy-compat")?\\s*?>)${isFlexible ? '(.+)?' : ''}`, 'gi');
```

Comment:
> 'about:legacy-compat'

Both the use of `"` and `'` are valid (see: https://html.spec.whatwg.org/multipage/syntax.html#the-doctype).

Reply:
@CKGrafico this too ⬆